### ISSUE TITLE
[BUG] Can't delete nodes: Kubernetes node missing: node has been removed from the cluster and there is no manager pod running on it

### ISSUE THREAD

**Issue by blampe** [longhorn/longhorn#5151]
## Describe the bug (🐛 if you encounter this issue)

If a node is removed from the Kubernetes cluster while Longhorn is not running, when Longhorn comes back it is unable to remove the now-missing node:

The node.longhorn.io will stay in a NotReady state with the status message

> Kubernetes node missing: node [...] has been removed from the cluster and there is no manager pod running on it

## To Reproduce

1. Scale Longhorn deployments to 0.
2. Remove a node from the Kubernetes cluster.
3. Scale Longhorn back to normal.
4. Attempt to delete the corresponding node.longhorn.io.

## Expected behavior

The Longhorn node should be deletable.

## Log or Support bundle

If applicable, add the Longhorn managers' log or support bundle when the issue happens. 
You can generate a Support Bundle using the link at the footer of the Longhorn UI.

## Environment

 - Longhorn version: 1.4.0-rc2
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): helm
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: k3s 1.25.4
   - Number of management node in the cluster: 
   - Number of worker node in the cluster: 
 - Node config
   - OS type and version:
   - CPU per node:
   - Memory per node:
   - Disk type(e.g. SSD/NVMe):
   - Network bandwidth between the nodes:
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): Baremetal
 - Number of Longhorn volumes in the cluster:

**Comment by joshimoo** [comment]
@longhorn/qa is this a potential regression in 1.4?

**Comment by innobead** [comment]
The related code is for a while, and it looks like the node should be deleted after the next reconciling (30 seconds).  

https://github.com/longhorn/longhorn-manager/blob/9d57b4cdd29b7997eee65b7cea2ad8d6f0f3a19b/controller/node_controller.go#L355-L358

Don't think we have any delete validation hook currently. cc @derekbit

**Comment by derekbit** [comment]
> Don't think we have any delete validation hook currently. cc @derekbit

Right. We don't have node deletion validatoin https://github.com/longhorn/longhorn-manager/blob/v1.4.0-rc2/webhook/resources/node/validator.go#L36.

**Comment by khushboo-rancher** [comment]
This scenario is part of QA's release testing, we didn't observe the mentioned issue. Also, I once again tried deletion of node with Longhorn v1.4.0-rc2, things are looking fine.

>Kubernetes node missing: node has been removed from the cluster and there is no manager pod running on it.

This could be an intermittent error shown on UI page, this will go after sometime and the node will get deleted successfully from Longhorn.

Also, answers of https://github.com/longhorn/longhorn/issues/5151#issuecomment-1366321254 questions can help to understand this better.

**Comment by blampe** [comment]
> Scale Longhorn deployments to 0.. What deployment in the step?

I had drained the nodes LH usually runs on, so LH was not running at the time.

> How did you delete the longhorn node? by UI or by kubectl?

kubectl 

The issue only resolved after I manually removed nodes/UPDATE from the validating & mutating LH webhooks.
